src/store/modules/wallets/wallets-plugin.ts

```diff
@@ -44,14 +40,21 @@ const walletsPersistencePlugin = new VuexPersistence({
   key: 'adm-wallets',
   storage: window.localStorage,
   restoreState: (key, storage) => {
-    const wallets = JSON.parse(storage.getItem(key))
-    console.log('restore state')
+    let wallets = {}
+    if (storage) {
+      const item = storage.getItem(key)
+      if (item) {
+        wallets = JSON.parse(item)
+      }
+    }
+
     return {
       wallets
     }
   },
-  saveState: (key, state, storage) => {
-    console.log('save state', state.wallets.symbols)
+  saveState: (key, state: Record<string, WalletsState>, storage) => {
+    if (!storage) throw new Error('Storage is undefined in wallets plugin')
```

Comment:
No, in vuex-persist storage has type Storage | undefined. I cannot just type it as a Storage, otherwise saveState will throw type error on build. That's why I did this check

Reply:
I see, the types of VuexPersistence are poor. Then use: `storage!.getItem(key)` and `storage!.setItem()`. It must be safe because the storage is always defined.